docs/book/v1/migrate-implements-factory-interface-to-use-psr-container.md

```diff
@@ -0,0 +1,114 @@
+use Interop\Container\ContainerInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+
+class ServiceFactory
+{
+    public function __invoke(\Psr\Container\ContainerInterface $container)
+    {
+        return new Service();
+    }
+}
+```
+
+The steps to apply the changes are:
+
+- Remove the interface from Factory class
+- Replace `invoke__()` class method param and the use of Interop class in the class
+
+To apply that, we can register rector rule: `Laminas\ServiceManager\Migration\Rector\Class_\ImplementsFactoryInterfaceToPsrFactoryRector` to our `rector.php` as an individual service:
+
+```php
+use Rector\Core\Configuration\Option;
+use Laminas\ServiceManager\Migration\Rector\Set\ValueObject\SetList;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+
+return static function (ContainerConfigurator $containerConfigurator): void {
+    $containerConfigurator->import(SetList::LAMINAS_SERVICEMANGER_40);
+
+    $parameters = $containerConfigurator->parameters();
+    $parameters->set(Option::PATHS, [__DIR__ . '/module']);
+
+     // register ImplementsFactoryInterfaceToPsrFactoryRector service
+    $services = $containerConfigurator->services();
+    $services->set(
+        \Laminas\ServiceManager\Migration\Rector\Class_\ImplementsFactoryInterfaceToPsrFactoryRector::class
+    );
+};
+```
+
+If we want to auto import, we can use `LAMINAS_SERVICEMANGER_40_AUTO_IMPORT`:
+
+```php
+use Rector\Core\Configuration\Option;
+use Laminas\ServiceManager\Migration\Rector\Set\ValueObject\SetList;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+
+return static function (ContainerConfigurator $containerConfigurator): void {
+    $containerConfigurator->import(SetList::LAMINAS_SERVICEMANGER_40_AUTO_IMPORT);
+
+    $parameters = $containerConfigurator->parameters();
+    $parameters->set(Option::PATHS, [__DIR__ . '/module']);
+
+    // register ImplementsFactoryInterfaceToPsrFactoryRector service
+    $services = $containerConfigurator->services();
+    $services->set(
+        \Laminas\ServiceManager\Migration\Rector\Class_\ImplementsFactoryInterfaceToPsrFactoryRector::class
+    );
+};
+```
+
+After configuration in place, we can run:
+
+```bash
+vendor/bin/rector process --dry-run
+```
+
+Ensure that the change is correct, if everything ok, we can run the fix:
+
+```bash
+vendor/bin/rector process
+```
+
+## Additional Adjustment
+
+- Add Return type for the service, for example:
+
+```php
+use My\Service;
+use Psr\Container\ContainerInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+
+class ServiceFactory
+{
+    public function __invoke(ContainerInterface $container): Service
+    {
+        return new Service();
+    }
+}
+```
+
+For add return type by new instance creation, we can use [`ReturnTypeFromReturnNewRector` TypeDeclaration rule](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#returntypefromreturnnewrector).
```

Comment:
A short code example like above would be great.

Reply:
updated code before after like above https://github.com/laminas/laminas-servicemanager-migration/pull/35/commits/3c0c1caa1b3af4ab5d97078da8e9bd11ca828c12